src/OrbitGl/CoreMathTest.cpp

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <gtest/gtest.h>
+
+#include "CoreMath.h"
+
+namespace orbit_gl {
+
+ClosedInterval kFakeClosedInterval{0, 3};
+
+TEST(ClosedInterval, FromValues) {
+  EXPECT_EQ(kFakeClosedInterval, ClosedInterval::FromValues(0, 3));
+  EXPECT_EQ(kFakeClosedInterval, ClosedInterval::FromValues(3, 0));
+}
+
+TEST(ClosedInterval, Intersects) {
+  EXPECT_TRUE((ClosedInterval{0, 3}.Intersects(ClosedInterval{1, 2})));
+  EXPECT_TRUE((ClosedInterval{1, 2}.Intersects(ClosedInterval{0, 3})));
+  EXPECT_TRUE((ClosedInterval{0, 1}.Intersects(ClosedInterval{1, 2})));
+  EXPECT_TRUE((ClosedInterval{1, 2}.Intersects(ClosedInterval{0, 1})));
+  EXPECT_FALSE((ClosedInterval{0, 1}.Intersects(ClosedInterval{2, 3})));
+  EXPECT_FALSE((ClosedInterval{2, 3}.Intersects(ClosedInterval{0, 1})));
+}
+
+TEST(ClosedInterval, Contains) {
+  EXPECT_FALSE(kFakeClosedInterval.Contains(-1));
+  EXPECT_TRUE(kFakeClosedInterval.Contains(0));
+  EXPECT_TRUE(kFakeClosedInterval.Contains(1));
+  EXPECT_TRUE(kFakeClosedInterval.Contains(3));
+  EXPECT_FALSE(kFakeClosedInterval.Contains(4));
+}
+
+TEST(CoreMath, IsInsideRectangle) {
+  const Vec2 kFakeRectangleTopLeft{5, 5};
+  const Vec2 kFakeRectangleSize{2, 2};
+  EXPECT_FALSE(IsInsideRectangle({4, 4}, kFakeRectangleTopLeft, kFakeRectangleSize));
```

Comment:
Do we really need so many test cases 😂 ?

Reply:
For the FALSE we have one case in the top left, one in the bottom right, one at the top, one at the bottom, one at the left and one at the right.

For the TRUE, we have one per side + one in the center.

Reply:
I'm fine to keep it as now.

What I mean is, here our rectangle is a closure. So from my mind, I can only see three cases:
- nodes on the boundary
- nodes inside the closure
- nodes out side the closure

For the TRUE case,  you test four corners and the center node. But actually there are also some other nodes on the rectangle boundary. If we list it in that case, we can also test the center point of the left edge, and the center point of the top edge of the rectangle.

Similarly, for the FALSE case, we test cases for points outside the closure. Then in addition to the 6 cases you listed above, there are also bottom left and top right.
But this is just my personal preference 😂